### PR TITLE
first stab on fixing #1170

### PR DESCRIPTION
**Pull request by tarzanek**
#1170

### CODE REVIEW

**Comment by tarzanek** [comment]
basically the only two things missing are: 
1) pom.xml integration
2) position incrementor as in https://issues.apache.org/jira/secure/attachment/12663628/LUCENE-5897.patch   - this needs to be tested, junit do some of tests, so let's see if they pass, I will run also helius farm job to get full checks

**Comment by vladak** [comment]
Couple of comments:
  - turn this on for javascript analyzer
  - let xref to be generated
  - fix the Maven somehow
  - produce some sort of message when the file is skipped

File: src/org/opensolaris/opengrok/analysis/plain/PlainSymbolTokenizer.lex [hunk 1]
```diff
@@ -28,6 +28,7 @@ import org.opensolaris.opengrok.analysis.JFlexTokenizer;
 %class PlainSymbolTokenizer
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/analysis/plain/PlainSymbolTokenizer.lex]: copyright

File: src/org/opensolaris/opengrok/analysis/plain/PlainFullTokenizer.lex [hunk 2]
```diff
@@ -30,6 +30,7 @@ import org.opensolaris.opengrok.analysis.JFlexTokenizer;
 %class PlainFullTokenizer
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/analysis/plain/PlainFullTokenizer.lex]: copyright

File: src/org/opensolaris/opengrok/analysis/javascript/JavaScriptSymbolTokenizer.lex [hunk 3]
```diff
@@ -35,6 +35,7 @@ import org.opensolaris.opengrok.analysis.JFlexTokenizer;
 %class JavaScriptSymbolTokenizer
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/analysis/javascript/JavaScriptSymbolTokenizer.lex]: copyright

File: src/org/opensolaris/opengrok/analysis/java/JavaSymbolTokenizer.lex [hunk 4]
```diff
@@ -36,6 +36,7 @@ import org.opensolaris.opengrok.analysis.JFlexTokenizer;
 super(in);
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/analysis/java/JavaSymbolTokenizer.lex]: copyright

File: opengrok-indexer/pom.xml [hunk 5]
```diff
@@ -131,6 +131,32 @@ Copyright (c) 2010, 2011, Oracle and/or its affiliates. All rights reserved.
                     </execution>
                 </executions>
             </plugin>
+<!-- TODO add the same fix as is in build.xml to patch jflex generated files to stop increasing buffer beyond token size that lucene accepts
```
Comment (tulinkry) [inline opengrok-indexer/pom.xml]: maybe copyright?

**Comment by tarzanek** [comment]
@tulinkry , copyright fixed ... will you have time to add the regexps to pom.xml ?

**Comment by tarzanek** [comment]
let's merge this, let the fun begin :-D

**Comment by tarzanek** [comment]
fwiw - 
fix the Maven somehow - let's sort this out in next pull req
produce some sort of message when the file is skipped - file won't be skipped, just the buggy token, which I think is safe to ignore, I am still afraid the index size and search speed will be impacted by this, since we might get 1MB doc fields thanks to this ...

**Pull request merged**